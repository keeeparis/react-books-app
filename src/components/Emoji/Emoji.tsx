import React, { FC } from 'react'

interface EmojiProps {
  label?: string
  symbol: any
}

const Emoji: FC<EmojiProps> = (props) => {
  return (
    <span
      className="emoji"
      role="img"
      aria-label={props.label ? props.label : ''}
      aria-hidden={props.label ? 'false' : 'true'}
    >
      {props.symbol}
    </span>
  )
}

export default Emoji
