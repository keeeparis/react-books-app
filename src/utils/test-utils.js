import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'
import 'regenerator-runtime/runtime'
import { Provider } from 'react-redux'
import { store } from '../redux/store/store'
import { MemoryRouter } from 'react-router-dom'
import '../i18n/config'

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  )
}

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { userEvent }
export * from 'regenerator-runtime/runtime'
export * from '../i18n/config'

export { customRender as render }
