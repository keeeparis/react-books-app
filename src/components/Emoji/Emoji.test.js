import { render, screen, fireEvent, userEvent } from 'test-utils'
import Emoji from './Emoji'

const props = {
  label: 'books',
  symbol: 'ðŸ“š',
}

it('test Emoji Component', () => {
  const { asFragment } = render(<Emoji {...props} />)

  expect(asFragment()).toMatchSnapshot()
})
