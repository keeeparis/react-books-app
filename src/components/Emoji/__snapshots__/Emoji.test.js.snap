// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Emoji Component 1`] = `
<DocumentFragment>
  <span
    aria-hidden="false"
    aria-label="books"
    class="emoji"
    role="img"
  >
    ðŸ“š
  </span>
</DocumentFragment>
`;
